kratos/utilities/model_part_operation_utilities.cpp

```diff
@@ -176,45 +177,50 @@ void ModelPartOperationUtilities::SetCommunicator(
     rOutputModelPart.SetCommunicator(p_output_communicator);
 }
 
-ModelPart& ModelPartOperationUtilities::CreateOutputModelPart(
-    const std::string& rOutputSubModelPartName,
+void ModelPartOperationUtilities::FillOutputSubModelPart(
+    ModelPart& rOutputSubModelPart,
     ModelPart& rMainModelPart,
     std::vector<ModelPart::NodeType*>& rOutputNodes,
     std::vector<ModelPart::ConditionType*>& rOutputConditions,
     std::vector<ModelPart::ElementType*>& rOutputElements)
 {
-    KRATOS_ERROR_IF(rMainModelPart.HasSubModelPart(rOutputSubModelPartName))
-        << "\"" << rOutputSubModelPartName << "\" already exists in the \""
-        << rMainModelPart.FullName() << "\".\n";
-
-    // create the output sub model part
-    auto& r_output_model_part = rMainModelPart.CreateSubModelPart(rOutputSubModelPartName);
+    // check if the sub model part is empty. Here we use the data communicator
+    // of rMainModelPart assuming the communicators are not set yet in rOutputSubModelPart
+    // since it is assumed to be empty in all contaienrs.
+    const auto& r_data_communicator = rMainModelPart.GetCommunicator().GetDataCommunicator();
+    const auto& entity_info = r_data_communicator.SumAll(std::vector<unsigned int>{
```

Comment:
`MaxAll` would be more appropriate here (`OrAll` would be best, but we don't have that)

Reply:
In this case, both would result the same.